discv5/discv5-wire.md

```diff
@@ -208,7 +208,9 @@ sent as responses to a single query. Implementations may place a limit on the al
 maximum for `total`. If exceeded, additional responses may be ignored.
 
 When handling NODES as a response to FINDNODE, the recipient should verify that the
-received nodes match the requested distances.
+received nodes match the requested distances. 
+When handling NODES as a response to FINDNODE, the recipient might verify the validity of the ENR records. 
```

Comment:
I think this text sounds a bit robotic because the 'When handling NODES...` part is repeated. Also, the wording is weird: saying it `might verify` is too weak. It `must verify`, and specifically what needs to be verified is the signature and well-formedness of the ENR encoding.

Reply:
Not sure how we should phrase this best, TBH. I think that there are two paths that implementations can take:

- (1) verify the signature, etc. on everything when they receive it. This is OK, but kind of expensive in practice.
- (2) verify lazily once the implementation determines that the record is actually needed.

I do think that this section should convey the intended semantics better: when an invalid record is received in the NODES list, it should not lead to the entire response being ignored.

Reply:
Hey sorry, I missed the discussion on this. 

Why should implementations not drop the entire response?

My thinking has been if some node is sending us junk (invalid ENRs) the node is either faulty or malicious. My implementation drops the rest of the list and I was even going to ban the node that sent me the junk. The reason I dont ban the nodes, is because we can get some data loss over UDP and potentially its an innocent mistake.

What is the case for continuing to read further nodes from a malicious/faulty node? I guess it gives us opportunity for more responses but that just seems like an implementation decision/optimization rather than something that needs to be specified in the spec.

Reply:
In our Nim implementation we also drop the entire response on an invalid ENR. In order not to be verifying a whole list of potentially invalid ENRs. (Yes, verifying the signatures of the ENRs returned by the FindNodes requests is probably the most costly thing going on in discv5. Depends on the amount of requests of course.)

We do continue to read the list when an ENR gets filtered because of no valid ip, not the distance requested or a duplicate in the same response.

Reply:
I'm also not sure anymore. We ended up here because @asanso asked me in Discord about the correct behavior in this case.

My thinking was that implementation bugs may cause invalid nodes to be returned. go-ethereum does it sometimes.
But I don't want to specify some behavior that you guys don't agree with in your implementation, so it's also OK for me to just close this PR.